prov/rxm/src/rxm.h

```diff
@@ -371,8 +372,62 @@ void rxm_pkt_init(struct rxm_pkt *pkt);
 int rxm_ep_msg_mr_regv(struct rxm_ep *rxm_ep, const struct iovec *iov,
 		       size_t count, uint64_t access, struct fid_mr **mr);
 void rxm_ep_msg_mr_closev(struct fid_mr **mr, size_t count);
-struct rxm_buf *rxm_buf_get(struct rxm_buf_pool *pool);
-void rxm_buf_release(struct rxm_buf_pool *pool, struct rxm_buf *buf);
+
+#define RXM_BUF_GET(mr_local, pool)				\
+	(mr_local ? rxm_buf_get_ex(pool) : rxm_buf_get(pool))
+
+#define RXM_TX_BUF_GET(rxm_ep)						\
+	(struct rxm_tx_buf *)RXM_BUF_GET((rxm_ep)->msg_mr_local,	\
+					 &(rxm_ep)->tx_pool)
+
+#define RXM_RX_BUF_GET(rxm_ep)						\
+	(struct rxm_rx_buf *)RXM_BUF_GET((rxm_ep)->msg_mr_local,	\
+					 &(rxm_ep)->rx_pool)
+
+static inline
+struct rxm_buf *rxm_buf_get(struct rxm_buf_pool *pool)
+{
+	struct rxm_buf *buf;
+
+	fastlock_acquire(&pool->lock);
+	buf = util_buf_alloc(pool->pool);
+	if (OFI_UNLIKELY(!buf)) {
+		fastlock_release(&pool->lock);
+		return NULL;
+	}
+	memset(buf, 0, sizeof(*buf));
```

Comment:
It would be nice (as a future patch) to avoid initializing the buffer to 0 (here and below).  I realize this could result in some hard to debug issues if a field misses being set.   We could also see if setting the fields to 0 on put would hide the costs more when running latency tests.  (I'm not sure if that would be the case).

Reply:
Yes, it would be nice. I thought for a while how we can do that. The first idea that arises is to prepare separate pools for TX(tagged/msg)/RX buffers. Seems I've mentioned this idea in the previous review round.

Reply:
It seems this memset() is not needed at all. I haven't found any places where it's needed
@a-ilango What do you think?

Reply:
I think the memset can be removed. It might make debugging harder in rare cases.

> The first idea that arises is to prepare separate pools for TX(tagged/msg)/RX buffers. Seems I've mentioned this idea in the previous review round.

Yes, I think having separate pools would help.